src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -231,6 +268,13 @@ internal static string GenerateDataFieldName(ImmutableArray<byte> data)
 
         private static char Hexchar(int x)
             => (char)((x <= 9) ? (x + '0') : (x + ('A' - 10)));
+
+        private sealed class FieldComparer : IComparer<SynthesizedStaticField>
+        {
+            public static readonly FieldComparer Instance = new FieldComparer();
+
+            public int Compare(SynthesizedStaticField x, SynthesizedStaticField y) => x.Name.CompareTo(y.Name);
```

Comment:
Based on the way it's used, not handling null seems acceptable, but there should probably be a comment.